pkg/workflow/group_pool_wf_test.go

```diff
@@ -0,0 +1,335 @@
+		Name:   param.NewString(volumeName),
+		Size:   &sizeField,
+		PoolId: param.NewString(*poolResp.ID),
+		Online: param.NewBool(false),
+	}
+	_, err := suite.volumeService.CreateVolume(newVolume)
+	assert.Nilf(suite.T(), err, "Unable to create volume %v", volumeName)
+}
+
+// Delete Volume
+func (suite *GroupPoolWorkflowSuite) deleteVolume(volumeName string) {
+	volObj, _ := suite.volumeService.GetVolumeByName(volumeName)
+	err := suite.volumeService.DeleteVolume(*volObj.ID)
+	assert.Nilf(suite.T(), err, "Unable to delete volume %v", volumeName)
+
+}
+
+// Get Pool Detail
+func (suite *GroupPoolWorkflowSuite) GetPoolDetail(poolName string) *nimbleos.Pool {
+	poolResp, err := suite.poolService.GetPoolByName(poolName)
+	assert.Nilf(suite.T(), err, "Unable to get %v pool details", poolName)
+	return poolResp
+}
+
+// Merge non-default Pool to default Pool
+func (suite *GroupPoolWorkflowSuite) TestMergePool() {
+	// Get Non default pool name from array (after group merge a new pool will be created)
+	filter := &param.GetParams{}
+	poolResp, _ := suite.poolService.GetPools(filter)
+	var poolCount int = len(poolResp)
+	var nonDefaulPoolName string = ""
+	var nonDefaulPoolID string = ""
+	for i := 0; i < poolCount; i++ {
+		if *poolResp[i].Name != "default" {
+			nonDefaulPoolName = *poolResp[i].Name
+			nonDefaulPoolID = *poolResp[i].ID
+			fmt.Printf("Non default Pool Name %+v\n", nonDefaulPoolName)
+			fmt.Printf("Non default Pool ID %+v\n", nonDefaulPoolID)
+		}
+	}
+	defaultPool := suite.GetPoolDetail("default")
+	defaultPoolID := *defaultPool.ID
+	fmt.Printf("Default Pool ID %+v\n", defaultPoolID)
+
+	// Pool Merge - Merge pool to 'default' pool
+	_, err := suite.poolService.MergePool(nonDefaulPoolID, defaultPoolID, nil)
+	assert.Nilf(suite.T(), err, "Pool Merge failed")
+}
+
+// Edit Pool - remove array from pool
+func (suite *GroupPoolWorkflowSuite) TestModifyPoolArray() {
+	// defaultPool, _ := suite.poolService.GetPoolByName("default")
+	defaultPool := suite.GetPoolDetail("default")
+	exisitingPoolArrayList := defaultPool.ArrayList
+	defaultPoolID := *defaultPool.ID
+	arrayCount := int(*defaultPool.ArrayCount)
+	fmt.Printf("Default Pool Array Count %+v\n", arrayCount)
+
+	// Get one arrayId to retain in pool. Others Arrays will be unassigned from pool
+	arrayID := *exisitingPoolArrayList[0].ID
+	arraylist := &nimbleos.NsArrayDetail{
+		ID:      param.NewString(arrayID),
+		ArrayId: param.NewString(arrayID),
+	}
+
+	var updateArrayList []*nimbleos.NsArrayDetail
+	updateArrayList = append(updateArrayList, arraylist)
+
+	updatePoolInput := &nimbleos.Pool{
+		ArrayList: updateArrayList,
+	}
+	_, err := suite.poolService.UpdatePool(defaultPoolID, updatePoolInput)
+	assert.Nilf(suite.T(), err, "Pool Array list update failed")
+	fmt.Printf("Updated pool array list\n")
+	time.Sleep(5 * time.Second)
+}
+
+// Check if the element is present in the provided list/array/slice
+func Contains(slice []string, val string) (int, bool) {
```

Comment:
I looked at volume_service_test as an example, and the helper functions are starting with lower case. So contains would probably be consistent with other test files.